core/connection.js

```diff
@@ -619,3 +621,32 @@ Blockly.Connection.prototype.setShadowDom = function(shadow) {
 Blockly.Connection.prototype.getShadowDom = function() {
   return this.shadowDom_;
 };
+
+/**
+ * @return {string} A human-readable description of this Connection.
+ */
+Blockly.Connection.prototype.describe = function() {
+  var msg;
+  var block = this.sourceBlock_;
+  if (!block) {
+    return 'Orphan Connection';
+  } else if (block.outputConnection === this) {
+    msg = 'Output Connection of ';
+  } else if (block.previousConnection === this) {
+    msg = 'Previous Connection of ';
+  } else if (block.nextConnection === this) {
+    msg = 'Next Connection of ';
+  } else {
+    for (var i = 0; i < block.inputList.length; ++i) {
+      var input = block.inputList[i];
```

Comment:
The JS way of looping through lists (assuming no false values) is:
for (var i = 0, input; input = block.inputList[i]; i++) {